otheatre/src/main/java/com/orange/otheatre/otheatre/controller/ManageEventsController.java

```diff
@@ -93,12 +109,31 @@ public String addEventSubmit(HttpServletRequest request,
         }
 
         try {
-             saveEvent.saveEvent(eventToAdd);
-        } catch ( Exception ex ) {
-            LOGGER.error(ex.getMessage());
+            if(!eligibility.checkTimeSlot(hallRepository.findByHallName(hallName).getHallName(),eventToAdd.getEventStartDate(),eventToAdd.getEventEndDate()) &&
+                    !eligibility.checkPlayLength(eventToAdd.getEventStartDate(),eventToAdd.getEventEndDate()) &&
+                    !eligibility.checkPlayEndTime(eventToAdd.getEventEndDate())) {
+                saveEvent.saveEvent(eventToAdd);
```

Comment:
Don't put business logic in controller. In controller you just receive an event and pass to service in order to add that event, never do business logic in controller.
In controller you should have:

eventService.createEvent(event);

In method createEvent from eventService you will have to do the business logic.

All the business logic should be made in service.